apache/superset
# feat(sqllab): SPA migration

### SUMMARY
Part 4-9 of SQLLab SPA migration work: [detail link](https://docs.google.com/document/d/1opyNuaw-CEAD9tG6fx_zqNcZwe3_soQgsnSJTP4mQkk/edit?usp=sharing)

- Change the url from /superset/sqllab to /sqllab
- Update frontend routes
- Update unit tests and cypress e2e tests
- Update redirectSQLLab logic

For the redirect logic, it will replace the SupersetClient.post redirection by react-router redirection with the location state for the post payload data. This means the redundant form_data serialize/deserialize logic has been discarded (which is also another performance gain). 

~~FYI: This commit does *not* include the delete work (i.e. old entrypoint related files, redirect legacy endpoint to new endpoint). This work will be handled in the next PR to minimize the regression rollback.~~ 

### TESTING INSTRUCTIONS

1. Make sure the SQL Lab page works as before but in the context of SPA (does not reload the page when navigating)
2. Make sure that /superset/sqllab redirects to /sqllab including additional query parameters

### ADDITIONAL INFORMATION
